### PR TITLE
Consolidate 6 CMakeLists.txt under include/* into main CMakeLists.txt

### PR DESCRIPTION
**Pull request by solvingj**
This one is a very clean consolidation and simplification.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -343,3 +342,13 @@ INSTALL(
     DESTINATION CMake
     COMPONENT dev
 )
+
+#Recursively list of all headers, with paths relative to include/tins
+FILE(GLOB_RECURSE INCLUDE_FILES RELATIVE include/tins "*.h")
```
Comment (mfontanini) [inline CMakeLists.txt]: Does the `RELATIVE` clause restrict the lookup to the `include/tins` directory or will this search for any `.h` files starting from the project's root?

Comment (solvingj) [reply CMakeLists.txt]: Indeed, it's project root, expression should now be "include/tins/*.h".

Comment (mfontanini) [reply CMakeLists.txt]: Yeah, that makes sense. Otherwise this will probably end up installing the googlemock headers globally.

Comment (solvingj) [reply CMakeLists.txt]: Have a look now, narrowed the scope and simplified.

**Comment by mfontanini** [comment]
Thanks!

**Pull request merged**